client/src/components/atoms/SearchableDropdown.tsx

```diff
@@ -1,135 +1,124 @@
 import React, { FunctionComponent, useEffect, useState } from "react";
-import ScrollWindow from "../atoms/ScrollWindow";
+
+import ScrollableDropdown from '../atoms/ScrollableDropdown'
+import Tag from "../atoms/Tag";
 import { TextField } from "../atoms/TextField";
-import { useComponentVisible } from "../utils/hooks";
 
 interface Props {
+  dropdownItems: Array<string>,
   initialText: string;
-  placeholderText: string;
-  searchPlaceholderText: string;
-  dropdownItems: Array<string>;
-  isErroneous: boolean;
   isDisabled: boolean;
   isEmpty?: boolean;
-  onChange: React.ChangeEventHandler<HTMLInputElement>;
-  onSelect: (item: string) => void;
+  isErroneous: boolean;
+  isTagDropdown?: boolean;
   noItemsAction: React.ReactNode;
+  onChange: React.ChangeEventHandler<HTMLInputElement>,
+  onSelect: (item: string) => void;
+  placeholderText: string;
+  searchPlaceholderText: string;
 }
 
 const SearchableDropdown: FunctionComponent<Props> = (props: Props) => {
```

Comment:
I reorganized alot of the messy behaviour logic.
See the [feature doc](https://www.notion.so/uwblueprintexecs/Edit-Request-Modals-a6c0d90b1dde41b2a09dd3cf3a2f507e#a0eadd8b9e6a4e4ba65c27ca3f7c37af) for behaviour specs